idunn/api/instant_answer.py

```diff
@@ -191,26 +164,35 @@ async def get_instant_answer_intention(intention, query: str, lang: str):
         return no_instant_answer(query=query, lang=lang)
 
     if len(places) == 1:
-        place_id = places[0].id
-        result = InstantAnswerResult(
-            places=places,
-            source=places_bbox_response.source,
-            intention_bbox=intention.filter.bbox,
-            maps_url=maps_urls.get_place_url(place_id),
-            maps_frame_url=maps_urls.get_place_url(place_id, no_ui=True),
-        )
-    else:
-        result = InstantAnswerResult(
-            places=places,
-            source=places_bbox_response.source,
-            intention_bbox=intention.filter.bbox,
-            maps_url=maps_urls.get_places_url(intention.filter),
-            maps_frame_url=maps_urls.get_places_url(intention.filter, no_ui=True),
-        )
+        return build_single_ia_answer(lang, query, places[0], intention.filter.bbox)
+
+    result = InstantAnswerResult(
+        places=places,
+        source=places_bbox_response.source,
+        intention_bbox=intention.filter.bbox,
+        maps_url=maps_urls.get_places_url(intention.filter),
+        maps_frame_url=maps_urls.get_places_url(intention.filter, no_ui=True),
+    )
 
     return build_response(result, query=query, lang=lang)
 
 
+def get_single_ia_datasource_priority(
+    is_france_query, fetch_bragi_tripadvisor, fetch_pj, fetch_bragi_osm
+) -> List[Union[Datasource, any]]:
+    if is_france_query:
+        return [
+            (Tripadvisor(), fetch_bragi_tripadvisor),
+            (PagesJaunes(), fetch_pj),
+            (Osm(is_wiki_filter=False), fetch_bragi_osm),
+        ]
+    return [
+        (Osm(is_wiki_filter=True), fetch_bragi_osm),
+        (Tripadvisor(), fetch_bragi_tripadvisor),
+        (Osm(is_wiki_filter=False), fetch_bragi_osm),
+    ]
```

Comment:
Awesome :+1: 

That's a lazy question, but did you have to compromise in order to simplify this?

Reply:
Yes, I have to instantiate a new object for each priority which is not great because there is already some instantiation (for example with pj_source).

Reply:
Oh yeah indeed, this means that we will open a new HTTP session with page jaunes for each request.

Maybe we should instantiate those datasources once outside of `get_single_ia_datasource_priority` and only return already existing instances from this function?